libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1678,17 +1678,10 @@ void TypeChecker::typeCheckFunctionGeneralChecks(
 	{
 		auto const& parameterNames = _functionType->parameterNames();
 
-		// Check for expected number of named arguments
-		if (parameterNames.size() != argumentNames.size())
-		{
-			m_errorReporter.typeError(
-				_functionCall.location(),
-				parameterNames.size() > argumentNames.size() ?
-				"Some argument names are missing." :
-				"Too many arguments."
-			);
-			return;
-		}
+		solAssert(
```

Comment:
Where is the check that allows you to turn this into an assertion?

Reply:
If we can assume that the parameter names list is the same lengths as the parameter types list, which we can with the newly added asserts in the c'tor, then line 1593 does this check already: https://github.com/ethereum/solidity/pull/5957/files#diff-9f2a45efaa7b75ab7ec620d3177cf01aL1593